tests/test_utils.py

```diff
@@ -178,3 +183,30 @@ def test_async_barrier_reset(self):
         assert barrier.count == 5
         barrier.reset()
         assert barrier.count == 0
+
+
+@pytest.mark.asyncio
+async def test_encoded_text_write():
+    # 1. Get temporary filename
+    with NamedTemporaryFile() as temp_file:
+        file_name = temp_file.name
+
+    # 2. Generate random string
+    init_str = (''.join(choice(ascii_uppercase) for i in range(100)))
+
+    # 3. Write chuncked decoded string into file
+    async with AsyncFileWriter(
+            current_loop(),
+            file_name,
+            access_mode='w+',
+            decode=codecs.decode) as file_writer:
```

Comment:
If I add `max_chunks=1` argument, your test hangs.
Please find out why and fix it. :wink:

Reply:
Hint: compare the timing of calls to `write()` method with execution of `_write()` function in the asyncio executor.

Reply:
Fixed